File: api/Server/Routes/ModelRoutes.py
from api.Contract.ApiResponse import ApiResponse
from api.Server.CoreApiServer import CoreApiServer


class ModelRoutes:
    """集中注册模型管理相关 HTTP 路由。"""

    SNAPSHOT_PATH: str = "/api/models/snapshot"
    UPDATE_PATH: str = "/api/models/update"
    ACTIVATE_PATH: str = "/api/models/activate"
    ADD_PATH: str = "/api/models/add"
    DELETE_PATH: str = "/api/models/delete"
    RESET_PRESET_PATH: str = "/api/models/reset-preset"
    REORDER_PATH: str = "/api/models/reorder"
    LIST_AVAILABLE_PATH: str = "/api/models/list-available"
    TEST_PATH: str = "/api/models/test"

    @classmethod
    def register(cls, core_api_server: CoreApiServer, model_app_service) -> None:
        core_api_server.add_json_route(
            "POST",
            cls.SNAPSHOT_PATH,
            lambda request: ApiResponse(
                ok=True, data=model_app_service.get_snapshot(request)
            ),
        )
        core_api_server.add_json_route(
            "POST",
            cls.UPDATE_PATH,
            lambda request: ApiResponse(
                ok=True, data=model_app_service.update_model(request)
            ),
        )
        core_api_server.add_json_route(
            "POST",
            cls.ACTIVATE_PATH,
            lambda request: ApiResponse(
                ok=True, data=model_app_service.activate_model(request)
            ),
        )
        core_api_server.add_json_route(
            "POST",
            cls.ADD_PATH,
            lambda request: ApiResponse(
                ok=True, data=model_app_service.add_model(request)
            ),
        )
        core_api_server.add_json_route(
            "POST",
            cls.DELETE_PATH,
            lambda request: ApiResponse(
                ok=True, data=model_app_service.delete_model(request)
            ),
        )
        core_api_server.add_json_route(
            "POST",
            cls.RESET_PRESET_PATH,
            lambda request: ApiResponse(
                ok=True, data=model_app_service.reset_preset_model(request)
            ),
        )
        core_api_server.add_json_route(
            "POST",
            cls.REORDER_PATH,
            lambda request: ApiResponse(
                ok=True, data=model_app_service.reorder_model(request)
            ),
        )
        core_api_server.add_json_route(
            "POST",
            cls.LIST_AVAILABLE_PATH,
            lambda request: ApiResponse(
                ok=True, data=model_app_service.list_available_models(request)
            ),
        )
        core_api_server.add_json_route(
            "POST",
            cls.TEST_PATH,
            lambda request: ApiResponse(
                ok=True, data=model_app_service.test_model(request)
            ),
        )

File: src/backend/api/routes/model-routes.ts
import type { ApiRouteContext } from "./route-types";

export function register_model_routes(context: ApiRouteContext): void {
  const models = context.services.models.service;
  context.postJson("/api/models/snapshot", () => models.get_snapshot());
  context.postJson("/api/models/update", (body) => models.update_model(body));
  context.postJson("/api/models/activate", (body) => models.activate_model(body));
  context.postJson("/api/models/add", (body) => models.add_model(body));
  context.postJson("/api/models/delete", (body) => models.delete_model(body));
  context.postJson("/api/models/reset-preset", (body) => models.reset_preset_model(body));
  context.postJson("/api/models/reorder", (body) => models.reorder_model(body));
  context.postJson("/api/models/list-available", (body) => models.list_available_models(body));
  context.postJson("/api/models/test", (body) => models.test_model(body));
}
